Check authorized imports in LocalPythonInterpreter constructor

jank opened the pull request:
Checking if authorized_imports exist was marked as TODO in the comment.
This MR implements this functionality and adds tests.
Also adds a test option to the makefile to skip the tests of DocStrings, which require API keys.

AngeloKiriakoulis commented on src/smolagents/local_python_executor.py:
```diff
@@ -1529,7 +1530,12 @@ def __init__(
         }
-        # TODO: assert self.authorized imports are all installed locally
+        # assert self.authorized imports are all installed locally
+        missing_modules = [imp for imp in self.authorized_imports if find_spec(imp) is None]
```
There's an issue here if `additional_authorized_imports` includes `"*"`. A check should be added to ensure `"*"` is working properly.

jank commented:
@AngeloKiriakoulis good catch on the wildcard import. I added a test case for the wildcard import and fixed the implementation. Ran make style on changed files.
